packages/lesswrong/server/resolvers/commentResolvers.ts

```diff
@@ -105,4 +115,83 @@ augmentFieldsDict(Comments, {
       return (post && post.contents && post.contents.version) || "1.0.0";
     },
   },
+  doppelComments: {
+    resolveAs: {
+      type: '[DoppelComment]',
+      resolver: async (dbComment, args, context: ResolverContext) => {
+        const {DoppelComments, Users, currentUser} = context
+        if (!userHasDoppelComments(currentUser)) return []
+        // This should always be a single user
+        const [user] = await getWithLoader(context, Users, "usersByIdsForDoppelComments", {}, "_id", dbComment.userId, {karma: 1})
+        if (!user) return []
+        if (user.karma < 10_000) return []
+        const doppelComments = await getWithCustomLoader(context, "doppelCommentsForComment", dbComment._id, async (commentIds: string[]) => {
+          const doppelComments = await DoppelComments.find({commentId: {$in: commentIds}}).fetch()
+          const doppelsByComment = groupBy(doppelComments, 'commentId')
+          return commentIds.map(commentId => doppelsByComment[commentId] ?? [])
+        })
+        if (doppelComments.length < 2) {
+          // We'll make 'em for next time
+          void [createDoppelComment(dbComment, user), createDoppelComment(dbComment, user)]
```

Comment:
I think you want `void Promise.all([...])`?

Reply:
what would that do for you?